csds-client/client/util.go

```diff
@@ -269,6 +273,200 @@ func printOutResponse(response *csdspb.ClientStatusResponse, fileName string) er
 			}
 			fmt.Printf("Config has been saved to %v\n", fileName)
 		}
+
+		// call visualize to enable visualization
+		if visualization {
+			if err := visualize(out, monitor); err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+// visualize calls parseXdsRelationship and use the result to visualize
+func visualize(config []byte, monitor bool) error {
+	graphData, err := parseXdsRelationship(config)
+	if err != nil {
+		return err
+	}
+	dot, err := generateGraph(graphData)
+	if err != nil {
+		return err
+	}
+
+	if !monitor {
+		url := "http://dreampuf.github.io/GraphvizOnline/#" + dot
+		if err := openBrowser(url); err != nil {
+			return err
+		}
+	}
+
+	// save dot to file
+	f, err := os.Create("config_graph.dot")
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+	_, err = f.Write([]byte(dot))
+	if err != nil {
+		return err
+	}
+	fmt.Println("Config graph has been saved to config_graph.dot")
+	return nil
+}
+
+// struct stores the nodes and edges maps of graph
+type GraphData struct {
+	nodes     []map[string]string
+	relations []map[string]*treeset.Set
+}
+
+// parseXdsRelationship parses relationship between xds and stores them in GraphData
+func parseXdsRelationship(js []byte) (GraphData, error) {
+	var data map[string]interface{}
+	err := json.Unmarshal(js, &data)
+	if err != nil {
+		return GraphData{}, err
+	}
+	lds := make(map[string]string)
+	rds := make(map[string]string)
+	cds := make(map[string]string)
+	ldsToRds := make(map[string]*treeset.Set)
+	rdsToCds := make(map[string]*treeset.Set)
```

Comment:
I think we should also handle cds to eds

Reply:
What about adding handler for cds to eds in the v3 PR? Since for v2, there should not be eds in the detailed config?

Reply:
SG.